integration/v3_grpc_test.go

```diff
@@ -491,3 +491,168 @@ func TestV3WatchCancel(t *testing.T) {
 
 	clus.Terminate(t)
 }
+
+func TestV3RangeRequest(t *testing.T) {
+	tests := []struct {
+		putKeys []string
+		reqs    []pb.RangeRequest
+
+		wresps [][]string
```

Comment:
Considered it, but the test becomes really clunky and winds up retesting kv stuff covered in the storage kv tests. I added a check for the header revision in the loop though.